### PR TITLE
Move common requirements to chainer/setup.py

### PR DESCRIPTION
**Pull request by niboshi**
Fixes chainer/chainer#8228
Requires chainer/chainer#8241

### CODE REVIEW

File: run_cupy_combination_test.py [hunk 1]
```diff
@@ -45,16 +45,7 @@
         'setuptools',
         pip_require,
         'cython==0.29.13'
-    ] + conf['requires'] + [
```
Comment (kmaehashi) [inline run_cupy_combination_test.py]: Cannot remove this for CuPy tests as CuPy does not use `pip install -e .[jenkins]`.

File: test.sh [hunk 2]
```diff
@@ -8,6 +8,8 @@ pip install --user -e cupy/
 
 cd chainer
 
+pip install --user -e .[jenkins]
```
Comment (kmaehashi) [inline test.sh]: Is it intentional to install Chainer twice?

File: run_test.py [hunk 3]
```diff
@@ -287,15 +287,6 @@
         'IDEEP': 'ideep4py' if use_ideep else 'none',
         'CHAINER_BUILD_CHAINERX': '1' if build_chainerx else '0',
     }
-    conf['requires'] += [
```
Comment (kmaehashi) [inline run_test.py]: ditto

File: run_cupy_combination_test.py (hunk 1 above)
Comment (niboshi) [reply run_cupy_combination_test.py]: Fixed in a442c9fd77874fa51bedae705411e573c6ffec86.

File: run_test.py (hunk 3 above)
Comment (niboshi) [reply run_test.py]: a442c9fd77874fa51bedae705411e573c6ffec86

File: test.sh (hunk 2 above)
Comment (niboshi) [reply test.sh]: Fixed in 38e50893122d24470d236c0475feab23445ebae7.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Blocked by https://github.com/chainer/chainer/pull/8252

**Comment by kmaehashi** [comment]
Hmm, tests using subprocesses expects that Chainer module is installed.

**Comment by niboshi** [comment]
I hope it's fixed with 24d69c3e1b76d5757e704d7e6bd90649c31db2c2.

**Comment by kmaehashi** [comment]
Thanks for the prompt fix! pfnCI, test this please.

**Comment by toslunar** [comment]
Resolved merge conflicts: https://github.com/chainer/chainer-test/compare/master...toslunar:move-common-requirements
May I push it?

**Comment by niboshi** [comment]
@toslunar 
Thanks! Please go ahead.

**Comment by kmaehashi** [comment]
It seems that `coverage` command is unavailable in PATH because it's installed as `--user`.

**Comment by toslunar** [comment]
Pushed the merge commit.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**